sycl/include/sycl/group_algorithm.hpp

```diff
@@ -124,9 +127,9 @@ struct is_complex
 
 // ---- is_arithmetic_or_complex
 template <typename T>
-using is_arithmetic_or_complex =
-    std::integral_constant<bool, sycl::detail::is_complex<T>::value ||
-                                     sycl::detail::is_arithmetic<T>::value>;
+using is_arithmetic_or_complex = std::integral_constant<
+    bool, sycl::detail::is_complex<typename std::remove_cv_t<T>>::value ||
```

Comment:
I would prefer that this is done in `is_complex` instead, as it should really support that a type is complex even if it is const and/or volatile qualified.

Reply:
An attempt to  modify `is_complex` by simply replacing  `T` to `std::remove_cv_t<T>` introduced errors from other conditions using it.
If that's fine, we can merge it in as is, I create another ticket to have it addressed later.

Reply:
@steffenlarsen 
having the functionality in `is_complex` as you suggested requires changing other macros such as 
```
template <typename T, typename BinaryOperation>
using is_plus_or_multiplies_if_complex = std::integral_constant<
    bool, (is_complex<T>::value ? (is_plus<T, BinaryOperation>::value ||
                                   is_multiplies<T, BinaryOperation>::value)
                                : std::true_type::value)>;
```
where `is_plus` and `is_multiplies` need to be modified to have their cv attributes removed.